mail_tracking/models/mail_thread.py

```diff
@@ -0,0 +1,141 @@
+        """Adds CC recipient to the message.
+
+        Because Odoo implementation avoid store 'from, to, cc' recipients we
+        ensure that this information its written into the mail.message record.
+        """
+        message_dict.update(
+            {
+                "email_cc": message_dict.get("cc", False),
+                "email_to": message_dict.get("to", False),
+            }
+        )
+        return super()._message_route_process(message, message_dict, routes)
+
+    def _routing_handle_bounce(self, email_message, message_dict):
+        bounced_message = message_dict["bounced_message"]
+        if bounced_message.mail_tracking_ids:
+            # TODO detect hard of soft bounce
+            bounced_message.mail_tracking_ids.event_create("soft_bounce", message_dict)
+        return super()._routing_handle_bounce(email_message, message_dict)
+
+    def _message_get_suggested_recipients(self):
+        """Adds email 'extra' recipients as suggested recipients.
+
+        If the recipient has a res.partner, use it.
+        """
+        res = super()._message_get_suggested_recipients()
+        self._add_extra_recipients_suggestions(res, "email_cc", _("Cc"))
+        self._add_extra_recipients_suggestions(res, "email_to", _("Anon. To"))
+        return res
+
+    def _add_extra_recipients_suggestions(self, suggestions, field_mail, reason):
+        ResPartnerObj = self.env["res.partner"]
+        aliases = self.env["mail.alias"].get_aliases()
+        email_extra_formated_list = []
+        for record in self:
+            emails_extra = record.message_ids.mapped(field_mail)
+            for email in emails_extra:
+                email_extra_formated_list.extend(email_split_and_format(email))
+        email_extra_formated_list = set(email_extra_formated_list)
+        email_extra_list = [x[1] for x in getaddresses(email_extra_formated_list)]
+        partners_info = self.sudo()._message_partner_info_from_emails(email_extra_list)
+        for pinfo in partners_info:
+            partner_id = pinfo["partner_id"]
+            email = email_split(pinfo["full_name"])[0].lower()
+            if not partner_id:
+                if email not in aliases:
+                    self._message_add_suggested_recipient(
+                        suggestions, email=email, reason=reason
+                    )
+            else:
+                partner = ResPartnerObj.browse(partner_id)
+                self._message_add_suggested_recipient(
+                    suggestions, partner=partner, reason=reason
+                )
+
+    @api.model
+    def _get_view(self, view_id=None, view_type='form', **options):
+        """Add filters for failed messages.
+
+        These filters will show up on any form or search views of any
+        model inheriting from ``mail.thread``.
+        """
+
+        arch, view = super()._get_view(view_id, view_type, **options)
+        if view_type in {"search", "form"}:
+            doc = etree.XML(arch)
```

Comment:
I face an error when accessing the Sales menu. Seems that you don't test before pushing your changes.
![image](https://github.com/OCA/social/assets/8724793/5de383eb-509a-4d4f-80df-b0d28e7259ba)

Reply:
It was a work commit. I fixed that error. The module is functional for almost everything.

Not all the problems are solved, e.g. it doesn't list all the error cases in the discuss/failed msg menu. This week I think the final version will be ready.